### PR TITLE
Tanium bug fixed

### PR DESCRIPTION
**Pull request by moishce**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/30907

## Description
 handle properly the '=', ';' characters wen input into "parameters".

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by ShahafBenYakir — COMMENTED** [review]
Not sure I understand the idea here (also ran and got the same results as before). Just use `param.split('=',1)` instead of `param.split('=')`

**Review by yaakovi — CHANGES_REQUESTED** [review]
add a unit test specific for this function.
see it fails for the original code and passes with the updated cde

File: Packs/Tanium/ReleaseNotes/1_0_4.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### Tanium v2
+- Fixed an issue that not handled properly the '=', ';' characters when input into the value on "parameters"
+- Upgraded the Docker image to demisto/python3:3.8.6.13358.
```
Comment (ShahafBenYakir) [inline Packs/Tanium/ReleaseNotes/1_0_4.md:5]:
```suggestion
- Fixed an issue where parameters were not handled properly if contained '=' or ';' characters,
- Upgraded the Docker image to demisto/python3:3.8.6.13358.
```

File: Packs/Tanium/Integrations/Tanium_v2/Tanium_v2.py [hunk 2]
```diff
@@ -78,11 +77,16 @@ def parse_sensor_parameters(self, parameters):
...
+            if '=' not in param or param.startswith('=') or param.endswith('='):
+                add_to_the_previous_pram = f';{param}'
+                continue
             parameter_conditions.append({
-                'key': param.split('=')[0],
-                'value': param.split('=')[1]})
-
+                'key': param.split('=', 1)[0],
+                'value': param.split('=', 1)[1]})
         return parameter_conditions
```
Comment (ShahafBenYakir) [inline Packs/Tanium/Integrations/Tanium_v2/Tanium_v2.py:90]: Let's reverse the list again (to correspond to the previous order)

File: Packs/Tanium/Integrations/Tanium_v2/Tanium_v2.py [hunk 3]
```diff
@@ -78,11 +77,16 @@ def parse_sensor_parameters(self, parameters):
     def parse_action_parameters(self, parameters):
```
Comment (ShahafBenYakir) [inline Packs/Tanium/Integrations/Tanium_v2/Tanium_v2.py:77]: Add doc string and type hints

File: Packs/Tanium/Integrations/Tanium_v2/Tanium_v2_test.py [hunk 4]
```diff
@@ -306,3 +307,20 @@ def test_get_question_result_invalid_input():
...
+    ('key1=value1=value1', [{'key': 'key1', 'value': 'value1=value1'}]),
+    ('key1=value1=value1;key2=value2', [{'key': 'key2', 'value': 'value2'}, {'key': 'key1', 'value': 'value1=value1'}]),
+    ('key1=value1=value1;key2=valu;e2', [{'key': 'key2', 'value': 'valu;e2'}, {'key': 'key1', 'value': 'value1=value1'}]),
+    ('key1=value1=value1;key2=ab=;c', [{'key': 'key2', 'value': 'ab=;c'}, {'key': 'key1', 'value': 'value1=value1'}])
+
+]
+
+
+@ pytest.mark.parametrize('parameters, accepted_result', data_test_parse_action_parameters)
+def test_parse_action_parameters(parameters, accepted_result):
```
Comment (ShahafBenYakir) [inline Packs/Tanium/Integrations/Tanium_v2/Tanium_v2_test.py:323]: Add given when then with cases

File: Packs/Tanium/Integrations/Tanium_v2/Tanium_v2_test.py [hunk 5]
```diff
@@ -306,3 +307,28 @@ def test_get_question_result_invalid_input():
...
+    """Tests parse_action_parameters function
+    Given
+        string which contains keys and values separated by '=' and ';'.
+    When
+        - When calling the "parse_action_parameters" function to extract it to a dictionary
+    Then
+        - validate that everything is extracted properly even if there is within the value "=" or ";"
```
Comment (ShahafBenYakir) [inline Packs/Tanium/Integrations/Tanium_v2/Tanium_v2_test.py:330]:
```suggestion
    Given
        A string representing a key=value list separated by ';'
    When
        - When calling the "parse_action_parameters" function to extract it to a dictionary
    Then
        - validate that everything is extracted properly even if there is within the value "=" or ";"
```

Comment (ShahafBenYakir) [reply Packs/Tanium/Integrations/Tanium_v2/Tanium_v2_test.py:330]: Let's add cases in the Given and Then

**Review by ShahafBenYakir — APPROVED** [review]

**Review by yaakovi — APPROVED** [review]

**Pull request merged**